jni_gen/third_party/ApiSummarizer/src/main/java/com/github/hegde/mahesh/apisummarizer/doclet/AnnotationVisitor.java

```diff
@@ -0,0 +1,96 @@
+package com.github.hegde.mahesh.apisummarizer.doclet;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.AnnotationValue;
+import javax.lang.model.element.AnnotationValueVisitor;
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.type.TypeMirror;
+
+public class AnnotationVisitor implements AnnotationValueVisitor<Object, Void> {
+
+  private final ElementBuilders builders;
+  AstEnv env;
+
+  public AnnotationVisitor(ElementBuilders builders) {
+    this.builders = builders;
+    this.env = builders.env;
+  }
+
+  @Override
+  public Object visit(AnnotationValue annotationValue, Void unused) {
+    return null;
+  }
+
+  @Override
+  public Object visitBoolean(boolean b, Void unused) {
+    return b;
+  }
+
+  @Override
+  public Object visitByte(byte b, Void unused) {
+    return b;
+  }
+
+  @Override
+  public Object visitChar(char c, Void unused) {
+    return c;
+  }
+
+  @Override
+  public Object visitDouble(double v, Void unused) {
+    return v;
+  }
+
+  @Override
+  public Object visitFloat(float v, Void unused) {
+    return v;
+  }
+
+  @Override
+  public Object visitInt(int i, Void unused) {
+    return i;
+  }
+
+  @Override
+  public Object visitLong(long l, Void unused) {
+    return l;
+  }
+
+  @Override
+  public Object visitShort(short i, Void unused) {
+    return i;
+  }
+
+  @Override
+  public Object visitString(String s, Void unused) {
+    return s;
+  }
+
+  @Override
+  public Object visitType(TypeMirror typeMirror, Void unused) {
+    return builders.typeUsage(typeMirror);
+  }
+
+  @Override
+  public Object visitEnumConstant(VariableElement variableElement, Void unused) {
+    // TODO: Do this properly
```

Comment:
This is not self-documenting. What is not proper about it? Please file a GitHub issue.

nit: punctuation.